### PR TITLE
(plugin) apps::backup::rubrik::restapi - enhance compliance mode, use public api instead internal one

### PR DESCRIPTION
**Pull request by OlivierKuziemko**
Use of api/v1/report/compliance_summary_sla instead of api/internal/report/{id}/chart 

Rubrik APIv1 doc:
"
api/v1/report/complaince_summary_sla
Returns the compliance summary information for all protected objects based on whether the last expected snapshot was successful. This requirement is based on the SLA Domain assigned to the objects.
With
snapshot-range : 
An SLA Domain-based range of snapshots. Compliance is calculated for snapshots in the range.
"

api/v1/report/compliance_summary_sla can give us 4 ranges of snapshots:
Last snapshot
Last 2 snapshots
Last 3 snapshots
All snapshots

Add percent value

### CODE REVIEW

File: apps/backup/rubrik/restapi/mode/compliance.pm [hunk 1]
```diff
@@ -28,7 +28,19 @@ use warnings;
...
+    
+    if ($self->{option_results}->{snapshot_range} eq 'LastSnapshot') {
+        $prefix_output = 'last snapshot';
+    } elsif ($self->{option_results}->{snapshot_range} eq 'Last2Snapshot') {
+        $prefix_output = 'last 2 snapshots';
+    } elsif ($self->{option_results}->{snapshot_range} eq 'Last3Snapshot') {
+        $prefix_output = 'last 3 snapshots';
+    } else {
+        $prefix_output = 'all snapshots';
+    }
```
Comment (Sims24) [inline apps/backup/rubrik/restapi/mode/compliance.pm:41]: Hum, not sure `$self->{option_results}->{snapshot_range}` can be used here.

Comment (Sims24) [reply apps/backup/rubrik/restapi/mode/compliance.pm:41]: Did you test it?

**Review by Sims24 — CHANGES_REQUESTED** [review]
Hey, I reviewed this one. It's a breaking change, we need to handle properly, we'll have a discussion internally and be back to you asap. 

Thanks for the contribution!s

**Comment by garnier-quentin** [comment]
The endpoint ```/api/v1/report/compliance_summary_sla``` is available since version 5.2. It's a problem for older version. 

Moreover, we should use the endpoint ```/api/v1/report/compliance_summary``` (It's the same metric no ? We don't break it): 
```
Returns the compliance summary information for all protected objects based on a time-based requirement of at least one snapshot in each 24 hour report period. This view ignores the policies assigned to protected objects through SLA Domains.
```

What do you think about it ?  I don't know if we can get the CDM version to choose between internal and api v1.

**Comment by OlivierKuziemko** [comment]
I didn't check for older api version -_-.

It's not the same metrics:
Here's the result for my Rubrik cluster:
For compliance_summary
```
https://XXX/api/v1/report/compliance_summary
{
  "totalProtected": 956,
  "numberOfInComplianceSnapshots": 435,
  "numberOfOutOfComplianceSnapshots": 514,
  "numberOfAwaitingFirstFullSnapshot": 7,
  "percentOfInComplianceSnapshots": 45.8,
  "percentOfOutOfComplianceSnapshots": 54.2,
  "updatedTime": "2022-07-22T08:38:47.273Z"
}
``` 
For compliance_summary_sla
```
https://XXX/api/v1/report/compliance_summary_sla?snapshot_range=AllSnapshots
{
  "totalProtected": 956,
  "numberOfInComplianceSnapshots": 951,
  "numberOfOutOfComplianceSnapshots": 5,
  "percentOfInComplianceSnapshots": 99.4,
  "percentOfOutOfComplianceSnapshots": 0.6,
  "updatedTime": "2022-07-22T08:38:47.273Z",
  "updatedStatus": "Updated",
  "snapshotRange": "AllSnapshots"
}
```
It's very different.

The metric given by api/internal/report/{id}/chart are similar to /api/v1/report/compliance_summary_sla?snapshot_range=AllSnapshots

We can get CDM version through https://XXX/api/v1/cluster/me

**Comment by garnier-quentin** [comment]
Could you provide me (by email) an access to the API to test it and develop it ?

**Comment by garnier-quentin** [comment]
it should be quite the same. its weird. 

I will close the PR because the plugin uses the internal api. I could create a mapping for api v1 but i'll need an access to do that. If you (or someone else) can provide me that, i will do the development

**Pull request closed without merging**